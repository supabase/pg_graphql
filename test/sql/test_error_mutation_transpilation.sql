begin;
    comment on schema public is '@graphql({"inflect_names": true})';

    create table public.account(
        id serial primary key,
        first_name varchar(255) not null check (first_name not like '%_%')
    );

    -- Second mutation is supposed to generate an exception
    select
      jsonb_pretty(
        graphql.resolve($$
          mutation {
            firstInsert: insertIntoAccountCollection(objects: [
              { firstName: "name" }
            ]) {
              records {
                id
                firstName
              }
            }

            secondInsert: insertIntoAccountCollection(objects: [
              { firstName: "another_name" }
            ]) {
              records {
                id
                firstName
              }
            }
          }
        $$)
      );

    select * from public.account;

rollback;
