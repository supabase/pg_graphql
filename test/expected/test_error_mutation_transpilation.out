begin;
    comment on schema public is '@graphql({"inflect_names": true})';
    create table public.account(
        id serial primary key,
        first_name varchar(255) not null check (first_name not like '%_%')
    );
    -- Second mutation is supposed to generate an exception
    select
      jsonb_pretty(
        graphql.resolve($$
          mutation {
            firstInsert: insertIntoAccountCollection(objects: [
              { firstName: "name" }
            ]) {
              records {
                id
                firstName
              }
            }

            secondInsert: insertIntoAccountCollection(objects: [
              { firstName: "another_name" }
            ]) {
              records {
                id
                firstName
              }
            }
          }
        $$)
      );
                                                   jsonb_pretty                                                   
------------------------------------------------------------------------------------------------------------------
 {                                                                                                               +
     "data": null,                                                                                               +
     "errors": [                                                                                                 +
         {                                                                                                       +
             "message": "new row for relation \"account\" violates check constraint \"account_first_name_check\""+
         }                                                                                                       +
     ]                                                                                                           +
 }
(1 row)

    select * from public.account;
 id | first_name 
----+------------
(0 rows)

rollback;
